[quant] AddReLU module

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24217 [quant] AddReLU module**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
AddReLU module

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/functional_relu.py:
```diff
@@ -0,0 +1,17 @@
+from torch.nn._intrinsic import AddReLU as NNAddReLU
+
+class AddReLU(torch.nn.Module):
+    _FLOAT_MODULE = NNAddReLU
```
I think `import torch.nn._intrinsic as nni` and use `nni.AddReLU` might be more clear. I'm changing to use this pattern in my pr as well.

Also, maybe a better name for this file? functional_relu seems to be confusing.

zafartahirov replied:
Suggestions for the name of the file?

zafartahirov replied:
Resolved in person -- decided to go with the current name for now

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/functional_relu.py:
```diff
@@ -0,0 +1,17 @@
+import torch
+from torch.nn._intrinsic import AddReLU as NNAddReLU
+
+class AddReLU(torch.nn.Module):
```
Also, to reuse from_float, I think you'll need to inherit from `nnq.Add`?

zafartahirov replied:
nnq.Add won't exist anymore -- we are depricating it in #23635

jerryzh168 reviewed (changes requested):
see comments

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "AddReLU module"


Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/__init__.py:
```diff
@@ -1,9 +1,12 @@
 # @lint-ignore-every PYTHON3COMPATIMPORTS
 
+from .functional_relu import AddReLU
```
I still don't like this name, maybe just torch_relu?

jerryzh168 commented on test/common_quantization.py:
```diff
@@ -454,20 +455,26 @@ def __init__(self, quantized=False):
         if quantized:
             self.mycat = nnq.QFunctional()
             self.myadd = nnq.QFunctional()
+            self.my_intrinsic_add = nni.quantized.modules.AddReLU()
```
nit: ModForWrapping -> ModelForWrapping?

jerryzh168 commented on test/common_quantization.py:
```diff
@@ -454,20 +455,26 @@ def __init__(self, quantized=False):
         new_mod = cls(quantized=True)
         new_mod.mycat = new_mod.mycat.from_float(mod.mycat)
         new_mod.myadd = new_mod.myadd.from_float(mod.myadd)
+        new_mod.my_intrinsic_add = new_mod.my_intrinsic_add.from_float(mod.my_intrinsic_add)
```
and why do we write a separate from_float rather than relying on the tool?

zafartahirov replied:
I don't remember the reason :/

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

jerryzh168 commented on torch/nn/_intrinsic/quantized/modules/torch_relu.py:
```diff
@@ -0,0 +1,19 @@
+from torch.nn.quantized.modules._base_modules import _BaseQuantizedModule
+
+
+class AddReLU(torch.nn.Module, _BaseQuantizedModule):
```
wait, why not a QFunctionalReLU?

jerryzh168 replied:
please create a FunctionReLU and FunctionalReLU (maybe rename them as well...)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

raghuramank100 reviewed (changes requested):
Please see comments

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

Commit:
Update on "[quant] AddReLU module"


Pull Request resolved: https://github.com/pytorch/pytorch/pull/24217

Differential Revision: [D16777880](https://our.internmc.facebook.com/intern/diff/D16777880)

zafartahirov commented:
Closing as outdated in favor of #26612
